cloud/clusterapi.go

```diff
@@ -233,19 +231,19 @@ func (ca *ClusterApi) CreatePharmerSecret() error {
 	}
 
 	if err = CreateSecret(ca.kc, "pharmer-certificate", ca.namespace, map[string][]byte{
-		"ca.crt":             cert.EncodeCertPEM(CACert(ca.ctx)),
-		"ca.key":             cert.EncodePrivateKeyPEM(CAKey(ca.ctx)),
-		"front-proxy-ca.crt": cert.EncodeCertPEM(FrontProxyCACert(ca.ctx)),
-		"front-proxy-ca.key": cert.EncodePrivateKeyPEM(FrontProxyCAKey(ca.ctx)),
-		"sa.crt":             cert.EncodeCertPEM(SaCert(ca.ctx)),
-		"sa.key":             cert.EncodePrivateKeyPEM(SaKey(ca.ctx)),
+		"ca.crt":             cert.EncodeCertPEM(ca.pharmerCertificates.CACert.Cert),
+		"ca.key":             cert.EncodePrivateKeyPEM(ca.pharmerCertificates.CACert.Key),
+		"front-proxy-ca.crt": cert.EncodeCertPEM(ca.pharmerCertificates.FrontProxyCACert.Cert),
+		"front-proxy-ca.key": cert.EncodePrivateKeyPEM(ca.pharmerCertificates.FrontProxyCACert.Key),
+		"sa.crt":             cert.EncodeCertPEM(ca.pharmerCertificates.ServiceAccountCert.Cert),
+		"sa.key":             cert.EncodePrivateKeyPEM(ca.pharmerCertificates.ServiceAccountCert.Key),
 	}); err != nil {
 		return err
 	}
 
 	if err = CreateSecret(ca.kc, "pharmer-etcd", ca.namespace, map[string][]byte{
```

Comment:
required only for do, linode and packets. we should not create these secrets for all other providers